### PR TITLE
Add holiday data file

### PR DESCRIPTION
**Pull request by jodastephen**
A flexible mechanism to add new holiday calendars, which could also be used to replace built in ones. No change is made to the existing algorithmic calendar generation, as it is effective and minimal.

INI and property files now allow a shorthand format where `key = value` can be replaced by `key`, which sets the value to an empty string.

`TestHelper` was enhanced to allow logging to be captured.

### CODE REVIEW

File: modules/collect/src/test/java/com/opengamma/strata/collect/TestHelper.java [hunk 1]
```diff
@@ -372,7 +376,7 @@ public static synchronized String caputureSystemOut(Runnable runner) {
    * <p>
    * For example:
    * <pre>
-   *  String sysErr = caputureSystemerr(() -> myCode);
+   *  String sysErr = caputureSystemErr(() -> myCode);
```
Comment (cjkent) [inline modules/collect/src/test/java/com/opengamma/strata/collect/TestHelper.java]: Typo

File: modules/collect/src/test/java/com/opengamma/strata/collect/TestHelper.java [hunk 2]
```diff
@@ -395,6 +399,55 @@ public static synchronized String caputureSystemErr(Runnable runner) {
...
+   * For example:
+   * <pre>
+   *  String log = caputureLog(Foo.class, () -> myCode);
+   * </pre>
+   * 
+   * @param loggerClass  the class defining the logger to trap
+   * @param runner  the lambda containing the code to test
+   * @return the captured output
+   */
+  public static synchronized List<LogRecord> caputureLog(Class<?> loggerClass, Runnable runner) {
```
Comment (cjkent) [inline modules/collect/src/test/java/com/opengamma/strata/collect/TestHelper.java]: Typo in method name.

**Comment by cjkent** [comment]
The typo is the only problem

**Comment by jodastephen** [comment]
Fixed typo

**Pull request merged**